src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryUnary.cs

```diff
@@ -432,12 +431,21 @@ public static DateTimeValue NumberToDateTime(EvalVisitor runner, EvalVisitorCont
 
         public static DateTimeValue NumberToDateTime(FormattingInfo formatInfo, IRContext irContext, NumberValue value)
         {
-            var n = value.Value;
-            var date = _epoch.AddDays(n);
+            try
+            {
+                var n = value.Value;
+                var date = _epoch.AddDays(n);
 
-            date = MakeValidDateTime(formatInfo.TimeZoneInfo, date);
+                date = MakeValidDateTime(formatInfo.TimeZoneInfo, date);
 
-            return new DateTimeValue(irContext, date);
+                return new DateTimeValue(irContext, date);
+            }
+            catch (ArgumentOutOfRangeException)
+            {
+                (var shortMessage, _) = ErrorUtils.GetLocalizedErrorContent(TexlStrings.ErrTextOutOfRange, formatInfo.CultureInfo, out _);
```

Comment:
>ErrTextOutOfRange [](http://example.com/codeflow?start=88&length=17)

what error does PA give in this scenario?

Reply:
PA has a larger limit. 
`The arguments to the Text function do not represent a valid date or time value`